src/worker/activities/local_activities.rs

```diff
@@ -135,17 +152,61 @@ impl LocalActivityManager {
         }
     }
 
-    /// Mark a local activity as having completed. Returns the information about the local activity
-    /// so the appropriate workflow instance can be notified of completion.
-    pub(crate) fn complete(&self, task_token: &TaskToken) -> Option<LocalInFlightActInfo> {
-        let info = self
+    /// Mark a local activity as having completed (pass or fail)
+    pub(crate) fn complete(
+        &self,
+        task_token: &TaskToken,
+        status: &LocalActivityExecutionResult,
+    ) -> LACompleteAction {
+        if let Some(info) = self
             .dat
             .lock()
             .outstanding_activity_tasks
-            .remove(task_token)?;
-        self.semaphore.add_permits(1);
-        self.complete_notify.notify_one();
-        Some(info)
+            .remove(task_token)
+        {
+            self.semaphore.add_permits(1);
+
+            match status {
+                LocalActivityExecutionResult::Completed(_) => {
+                    self.complete_notify.notify_one();
+                    LACompleteAction::Report(info)
+                }
+                LocalActivityExecutionResult::Failed(_) => {
+                    match &info.la_info.schedule_cmd.retry_policy {
+                        None => {
+                            // If there's no retry policy... don't retry
```

Comment:
In practice, do we ever expect lang to not provide a retry policy?

Reply:
A default should be provided

Reply:
Hmm.. lang doesn't provide a default right now.
What do the other SDKs do? What is the default policy for LAs?

Reply:
It's to retry forever I believe. We can check the defaults when impling lang side